src/util/bitset.h

```diff
@@ -0,0 +1,510 @@
+    /** Check if all bits are 0. */
+    constexpr bool None() const noexcept { return m_val == 0; }
+    /** Check if any bits are 1. */
+    constexpr bool Any() const noexcept { return m_val != 0; }
+    /** Return an object that iterates over all 1 bits (++ and * only allowed when != end()). */
+    constexpr Iterator begin() const noexcept { return Iterator(m_val); }
+    /** Return a dummy object to compare Iterators with. */
+    constexpr IteratorEnd end() const noexcept { return IteratorEnd(); }
+    /** Find the first element (requires Any()). */
+    constexpr unsigned First() const noexcept { return std::countr_zero(m_val); }
+    /** Find the last element (requires Any()). */
+    constexpr unsigned Last() const noexcept { return std::bit_width(m_val) - 1; }
+    /** Set this object's bits to be the binary AND between respective bits from this and a. */
+    constexpr IntBitSet& operator|=(const IntBitSet& a) noexcept { m_val |= a.m_val; return *this; }
+    /** Set this object's bits to be the binary OR between respective bits from this and a. */
+    constexpr IntBitSet& operator&=(const IntBitSet& a) noexcept { m_val &= a.m_val; return *this; }
+    /** Set this object's bits to be the binary AND NOT between respective bits from this and a. */
+    constexpr IntBitSet& operator/=(const IntBitSet& a) noexcept { m_val &= ~a.m_val; return *this; }
+    /** Set this object's bits to be the binary XOR between respective bits from this as a. */
+    constexpr IntBitSet& operator^=(const IntBitSet& a) noexcept { m_val ^= a.m_val; return *this; }
+    /** Check if the intersection between two sets is non-empty. */
+    constexpr bool Overlaps(const IntBitSet& a) noexcept { return m_val & a.m_val; }
+    /** Return an object with the binary AND between respective bits from a and b. */
+    friend constexpr IntBitSet operator&(const IntBitSet& a, const IntBitSet& b) noexcept { return I(a.m_val & b.m_val); }
+    /** Return an object with the binary OR between respective bits from a and b. */
+    friend constexpr IntBitSet operator|(const IntBitSet& a, const IntBitSet& b) noexcept { return I(a.m_val | b.m_val); }
+    /** Return an object with the binary AND NOT between respective bits from a and b. */
+    friend constexpr IntBitSet operator/(const IntBitSet& a, const IntBitSet& b) noexcept { return I(a.m_val & ~b.m_val); }
+    /** Return an object with the binary XOR between respective bits from a and b. */
+    friend constexpr IntBitSet operator^(const IntBitSet& a, const IntBitSet& b) noexcept { return I(a.m_val ^ b.m_val); }
+    /** Check if bitset a and bitset b are identical. */
+    friend constexpr bool operator==(const IntBitSet& a, const IntBitSet& b) noexcept = default;
+    /** Check if bitset a is a superset of bitset b (= every 1 bit in b is also in a). */
+    friend constexpr bool operator>=(const IntBitSet& a, const IntBitSet& b) noexcept { return (b.m_val & ~a.m_val) == 0; }
```

Comment:
`.Subset()` and `.Superset()`?

Without these comments, I would have to guess at the meanings (and I would have guessed wrong).